src/v/cloud_storage/cache_service.cc

```diff
@@ -62,26 +62,61 @@ static constexpr std::string_view tmp_extension{".part"};
 
 cache::cache(
   std::filesystem::path cache_dir,
+  size_t disk_size,
+  config::binding<double> disk_reservation,
   config::binding<uint64_t> max_bytes_cfg,
+  config::binding<std::optional<double>> max_percent,
   config::binding<uint32_t> max_objects) noexcept
   : _cache_dir(std::move(cache_dir))
+  , _disk_size(disk_size)
+  , _disk_reservation(std::move(disk_reservation))
   , _max_bytes_cfg(std::move(max_bytes_cfg))
+  , _max_percent(std::move(max_percent))
   , _max_bytes(_max_bytes_cfg())
   , _max_objects(std::move(max_objects))
   , _cnt(0)
   , _total_cleaned(0) {
     if (ss::this_shard_id() == ss::shard_id{0}) {
+        update_max_bytes(); // initialize _max_bytes
         _max_bytes_cfg.watch([this]() { update_max_bytes(); });
```

Comment:
we should probably also watch `_max_percent` and `_disk_reservation`

Reply:
100% we should. thanks, I missed this.